src/main/java/core/basesyntax/shop/service/impl/ParseShopDataImpl.java

```diff
@@ -0,0 +1,40 @@
+package core.basesyntax.shop.service.impl;
+
+import core.basesyntax.shop.service.ParseShopData;
+import core.basesyntax.shop.strategy.ShopStrategy;
+import java.util.Map;
+
+public class ParseShopDataImpl implements ParseShopData {
+    private ShopStrategy fruitShopStrategy;
+
+    protected ParseShopDataImpl() {
+        fruitShopStrategy = new ShopStrategy();
+    }
+
+    @Override
+    public boolean distribute(String table) {
+        table = table.replaceAll("type,\\w+,quantity\\n", "");
+        String[] csvArr = table.split("\\n");
+        for (String line : csvArr) {
+            String type = line.replaceAll("((^[bspr]).+)", "$2");
+            String item = line.replaceAll("([bspr],(\\w+),\\d+)", "$2");
+            int quantity = Integer.parseInt(
+                    line.replaceAll("(.+,(\\d+)$)", "$2"));
+            fruitShopStrategy.chooseStrategy(type,item,quantity);
+        }
+        return true;
+    }
```

Comment:
Seems the same problem as with Validator. U need to use enum here

Reply:
Do you mean operations "b", "s", "p", "r"? But they are always the same, there's no need to change them in future

Reply:
As we was told at theory lesson today, we need to use this as a strategy.